ops/internal/manage/generate.go

```diff
@@ -21,22 +21,51 @@ import (
 // GenerateChainArtifacts creates a chain config and genesis file for the chain at index idx in the given state file
 // using the given shortName (and optionally, name and superchain identifier).
 // It writes these files to the staging directory.
-func GenerateChainArtifacts(statePath string, wd string, shortName string, name *string, superchain *string, idx int, opDeployerVersion string) error {
+func GenerateChainArtifacts(
+	statePath string,
+	wd string,
+	shortName string,
+	name *string,
+	superchain *string,
+	idx int,
+	opDeployerVersion string,
+	opDeployerBinDir string,
+) error {
 	st, err := deployer.ReadOpaqueStateFile(statePath)
 	if err != nil {
 		return fmt.Errorf("failed to read opaque state file: %w", err)
 	}
-	l1contractsrelease, err := st.ReadL1ContractsLocator()
-	if err != nil {
-		return fmt.Errorf("failed to read L1 contracts release: %w", err)
+
+	var picker deployer.BinaryPicker
+	if opDeployerVersion == "" {
```

Comment:
The logic around what to do if the op deployer version is blank is very much a concern of the manage package, not the deployer package. As such I've pulled that logic in here.